org.lflang/src/org/lflang/generator/MainContext.java

```diff
@@ -0,0 +1,127 @@
+package org.lflang.generator;
+
+import java.util.Properties;
+import java.util.function.Function;
+
+import org.eclipse.xtext.util.CancelIndicator;
+
+import org.lflang.DefaultErrorReporter;
+import org.lflang.ErrorReporter;
+import org.lflang.FileConfig;
+import org.lflang.TargetConfig.Mode;
+import org.lflang.generator.IntegratedBuilder.ReportProgress;
+
+/**
+ * A {@code MainContext} is a context that is not nested in
+ * any other generator context. There is one {@code MainContext}
+ * for every build process.
```

Comment:
It would be good to add a note here to describe what the context is used for...

Reply:
That is true, more explanation was needed somewhere. In addition to adding emphasis to the fact that `MainContext` implements the `LFGeneratorContext` interface, I have added detail to the JavaDoc for `LFGeneratorContext`.